src/kern/npf_alg_pptp.c

```diff
@@ -536,14 +540,15 @@ pptp_tcp_translate(npf_cache_t *npc, npf_nat_t *nt, bool forw)
 
 		/* Save the server call ID. */
 		gre_state->call_id[SERVER_CALL_ID]= pptp_call_reply->hdr.call_id;
+		gre_state->flags |= GRE_STATE_SERVER_CALL_ID;
 
 		/*
 		 * If client and server call IDs have been seen, create
 		 * new GRE connection state entry.
 		 */
 		if (gre_state->call_id[CLIENT_CALL_ID] != 0 &&
```

Comment:
This check seems no longer valid either, if we take zero call IDs as valid.

Reply:
This is the translated call-id created by NPF's portmap. it's always non zero.

Reply:
There are more cases in `pptp_gre_destroy()`, `pptp_gre_destroy_state()`, ...

Reply:
since it's non zero, there is no need to change those functions.

Reply:
> This is the translated call-id created by NPF's portmap. it's always non zero.

Why add this unnecessary assumption (and checks)? It is also implicit; that is, it is acquired on GRE state creation and should be always set. So, we should simplify the code and remove the redundant checks.